### ISSUE TITLE
Resource list keyboard selection of all items does not trigger

### ISSUE THREAD

**Issue by alex-page** [Shopify/polaris-react#1319]
# Issue summary

Focusing on this box and using the enter or space bar key does not select the items below like how clicking works.

<img width="235" alt="Screen Shot 2019-04-16 at 10 45 05 AM" src="https://user-images.githubusercontent.com/19199063/56219425-b9021c80-6034-11e9-83a1-7306486caf09.png">

## Expected behavior

I should be able to interact with this checkbox using my keyboard.

## Actual behavior

What actually happens?

I tab to the element and cannot toggle the state using the keyboard.

## Steps to reproduce the problem

1. Open the storybook instance, resource list, all components
1. Tab to the checkbox and try triggering it using space or enter
1. Nothing happens

## Specifications

- Are you using the React components? (Y/N): Yes
- Polaris version number: v3
- Browser: Chrome
- Device: Macbook Pro
- Operating System: OSX
